CUDA.Dockerfile

```diff
@@ -0,0 +1,75 @@
+# Build argumnets
+ARG CUDA_VER=11.7
+ARG UBUNTU_VER=22.04
+
+# Download the base image
+FROM nvidia/cuda:${CUDA_VER}.0-devel-ubuntu${UBUNTU_VER}
+# you can check for all available images at https://hub.docker.com/r/nvidia/cuda/tags
+
+# Install as root
+USER root
+
+# Install dependencies
+RUN apt-get update
+RUN apt-get upgrade -y
+RUN apt-get autoremove -y && \
+    apt-get clean && \
+    rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/*
+    
+RUN apt-get update && \
+    DEBIAN_FRONTEND="noninteractive" apt-get install --yes \
+    --no-install-recommends \
+    bash \
+    bash-completion \
+    cmake \
+    curl \
+    git \
+    libopenblas-dev \
+    linux-headers-$(uname -r) \
+    nano \
+    python3 python3-dev python3-pip python-is-python3 \
+    sudo \
+    wget
+
+RUN apt-get autoremove -y && \
+    apt-get clean && \
+    rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/*
+
+# Add a user `${USERNAME}` so that you're not developing as the `root` user
+ENV USERNAME=ibm
+RUN echo "${USERNAME}"
+RUN useradd ${USERNAME} \
+    --create-home \
+    --shell=/bin/bash \
+    --user-group && \
+    echo "${USERNAME} ALL=(ALL) NOPASSWD:ALL" >> /etc/sudoers.d/nopasswd
+
+# Change to your user
+USER ${USERNAME}
+WORKDIR /home/${USERNAME}
+
+# Install python packages as your user
+RUN pip install --upgrade pip
+RUN pip install pybind11 scikit-build
+RUN pip install torch torchvision torchaudio --extra-index-url https://download.pytorch.org/whl/cu116
```

Comment:
Is there a way of making the pytorch repository used (ie. `.../cu116`) depend on the `CUDA_VER` argument? The [repository URLs](https://pytorch.org/get-started/locally/) seem to be relatively predictable, although not available for all available CUDA versions.

If it turns out to be not feasible or cumbersome, moving it to an argument and adding a note to the documentation stating that a specific version of pytorch is used should be enough - end goal is being able to accommodate different (and future) versions easily.

Reply:
for now, there is no direct way to do that as PyTorch is not compiled for all Cuda versions.
Maybe we can give only two options
cuda 10 defaults to 10.2 (for cards that do not support CUDA 11)
and cuda 11 defaults to 11.6

Reply:
Thanks for looking into it: can you add a
```
ARG PYTORCH_PIP_URL=https://download.pytorch.org/whl/cu116
```
argument (along with a note on the documentation at https://github.com/IBM/aihwkit/pull/403#discussion_r943583257) on replacing this argument if using a specific cuda version? This would avoid having some extra logic and make it easier to future-proof.

Reply:
Updated with mentioned changes